sourmash/sbt.py

```diff
@@ -121,18 +122,27 @@ class SBT(Index):
     Notes
     -----
     We use two dicts to store the tree structure: One for the internal nodes,
-    and another for the leaves (datasets).
+    and another for the leaves (datasets). The positions of all internal nodes are
+    smaller than of leaves. If pos is the position of a leaf, then its parent position
+    can be found with:
+        parent_position = int(math.floor((pos - 1) / d))
+    For example, if d=2, then the parent position 5 corresponds to the children at leaf
+    positions 11 and 12
     """
 
     def __init__(self, factory, d=2, storage=None):
         self.factory = factory
         self._nodes = {}
-        self._missing_nodes = set()
         self._leaves = {}
         self.d = d
         self.next_node = 0
         self.storage = storage
 
+    @property
+    def _missing_nodes(self):
+        return {i for i in range(max(self._nodes))
+                if i not in self._nodes and i not in self._leaves}
+
```

Comment:
This doesn't quite work, because `self._nodes` might be empty. (It worked during loading because then we had `max_node` to keep track.

An implementation that doesn't assume dense SBTs is below:
```suggestion
     # TODO: make it into a cached_property, clean it if new node is inserted                                                                     
     @property
     def _missing_nodes(self):
         all_nodes = set()
         current_level = set()

         for i in self._leaves:
             parent = self.parent(i)
             if parent is not None:
                 current_level.add(parent.pos)

         while current_level:
             previous_level = set()
             for i in current_level:
                 parent = self.parent(i)
                 if parent is not None:
                     previous_level.add(parent.pos)
             all_nodes.update(current_level)
             current_level = previous_level

         return all_nodes - set(self._nodes)
```

The issue is that it might be expensive to calculate this all the time, so probably the [`@cached_property`](https://github.com/dib-lab/sourmash/blob/baaa3107ae74ccbeb1838c8f1589cd7dfa66b0c7/sourmash/lca/lca_db.py#L15) decorator from the LCA db might be useful here too.

(and `@cached_property` should be in some other place in sourmash, because it is not LCA DB-specific =])